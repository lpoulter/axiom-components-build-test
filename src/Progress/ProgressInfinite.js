import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import { getArcProperties } from './utils';
import renderFilter from './_filter';
import RadialProgress from './RadialProgress';
import './ProgressInfinite.css';

export default class ProgressInfinite extends Component {
  static propTypes = {
    /** Color of the indicator */
    color: PropTypes.oneOf([
      'subtle',
      'white',
    ]),
    /** Size of the indicator */
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    /** Size in REM units */
    sizeRem: PropTypes.string,
  };

  static defaultProps = {
    color: 'subtle',
    size: 'small',
  }

  UNSAFE_componentWillMount = renderFilter

  render() {
    const {
      color,
      ...rest
    } = this.props;
    const classes = classnames('ax-progress-infinite', `ax-progress-infinite--${color}`);
    const {
      diameter,
      innerRadius,
      strokeWidth,
      strokeDasharray,
      strokeDashoffset,
    } = getArcProperties({ percent: 75 });

    return (
      <RadialProgress { ...rest } className={ classes } diameter={ diameter }>
        <circle
            className="ax-progress-infinite__arc"
            r={ innerRadius }
            style={ {
              stroke: `url(#ax-progress-infinite__filter--${ color })`,
              strokeDasharray,
              strokeDashoffset,
              strokeWidth,
            } } />
      </RadialProgress>
    );
  }
}
