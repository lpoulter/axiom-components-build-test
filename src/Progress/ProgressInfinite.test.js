import React from 'react';
import renderer from 'react-test-renderer';
import ProgressInfinite from './ProgressInfinite';

const getComponent = (props = {}) =>
  renderer.create(
    <ProgressInfinite { ...props } />
  );

describe('ProgressInfinite', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with color', () => {
    ['white', 'subtle'].forEach((color) => {
      it(color, () => {
        const component = getComponent({ color });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
