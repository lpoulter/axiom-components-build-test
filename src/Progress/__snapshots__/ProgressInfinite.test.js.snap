// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressInfinite renders with color subtle 1`] = `
<svg
  className="ax-progress-infinite ax-progress-infinite--subtle ax-radial-progress ax-radial-progress--small"
  style={Object {}}
  viewBox="0 0 24 24"
>
  <g
    transform="translate(12, 12) rotate(-90)"
  >
    <circle
      className="ax-progress-infinite__arc"
      r={10}
      style={
        Object {
          "stroke": "url(#ax-progress-infinite__filter--subtle)",
          "strokeDasharray": 62.83185307179586,
          "strokeDashoffset": 15.707963267948966,
          "strokeWidth": 4,
        }
      }
    />
  </g>
</svg>
`;

exports[`ProgressInfinite renders with color white 1`] = `
<svg
  className="ax-progress-infinite ax-progress-infinite--white ax-radial-progress ax-radial-progress--small"
  style={Object {}}
  viewBox="0 0 24 24"
>
  <g
    transform="translate(12, 12) rotate(-90)"
  >
    <circle
      className="ax-progress-infinite__arc"
      r={10}
      style={
        Object {
          "stroke": "url(#ax-progress-infinite__filter--white)",
          "strokeDasharray": 62.83185307179586,
          "strokeDashoffset": 15.707963267948966,
          "strokeWidth": 4,
        }
      }
    />
  </g>
</svg>
`;

exports[`ProgressInfinite renders with defaultProps 1`] = `
<svg
  className="ax-progress-infinite ax-progress-infinite--subtle ax-radial-progress ax-radial-progress--small"
  style={Object {}}
  viewBox="0 0 24 24"
>
  <g
    transform="translate(12, 12) rotate(-90)"
  >
    <circle
      className="ax-progress-infinite__arc"
      r={10}
      style={
        Object {
          "stroke": "url(#ax-progress-infinite__filter--subtle)",
          "strokeDasharray": 62.83185307179586,
          "strokeDashoffset": 15.707963267948966,
          "strokeWidth": 4,
        }
      }
    />
  </g>
</svg>
`;
